### ISSUE TITLE
kubic repo: "[MIRROR] libslirp-4.3.1-1.el8.x86_64.rpm: Downloading successful, but checksum doesn't match."

### ISSUE THREAD

**Issue by eriksjolund** [containers/podman#7440]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

The __kubic__ repo contains some internal checksum mismatch for the package __libslirp__.

I discovered that when I put the installation instructions for Centos 8:
https://podman.io/getting-started/installation
```
# CentOS 8
sudo dnf -y module disable container-tools
sudo dnf -y install 'dnf-command(copr)'
sudo dnf -y copr enable rhcontainerbot/container-selinux
sudo curl -L -o /etc/yum.repos.d/devel:kubic:libcontainers:stable.repo https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/CentOS_8/devel:kubic:libcontainers:stable.repo
sudo dnf -y install podman
```
in a Dockerfile (but with `sudo` removed)

**Steps to reproduce the issue:**
1.

```
$ cat /etc/centos-release
CentOS Linux release 8.2.2004 (Core) 
$ ls
Dockerfile
$ cat Dockerfile
FROM docker.io/library/centos:8
RUN dnf -y update && \
    dnf -y module disable container-tools && \
    dnf -y install 'dnf-command(copr)' && \
    dnf -y copr enable rhcontainerbot/container-selinux && \
    curl -L -o /etc/yum.repos.d/devel:kubic:libcontainers:stable.repo https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/CentOS_8/devel:kubic:libcontainers:stable.repo && \
    dnf -y install podman
$ podman build -t centospodman .
```

**Describe the results you received:**

The command `podman build -t centospodman .` failed. The relevant error seems to be

```
[MIRROR] slirp4netns-1.1.4-1.el8.x86_64.rpm: Downloading successful, but checksum doesn't match. Calculated: 44974ede47ec7db75852725c3c62b4a6d720025af39d0b426c7ca1d78698b7c8(sha256)  Expected: 824505ad6cb0fec0b5a38b488d3e331098fd4ffbd1d955629d1e827b8ed98c76(sha256) 
(21/23): podman-2.0.5-1.el8.x86_64.rpm                                                                   21 MB/s |  23 MB     00:01    
[MIRROR] slirp4netns-1.1.4-1.el8.x86_64.rpm: Downloading successful, but checksum doesn't match. Calculated: 44974ede47ec7db75852725c3c62b4a6d720025af39d0b426c7ca1d78698b7c8(sha256)  Expected: 824505ad6cb0fec0b5a38b488d3e331098fd4ffbd1d955629d1e827b8ed98c76(sha256) 
[MIRROR] slirp4netns-1.1.4-1.el8.x86_64.rpm: Downloading successful, but checksum doesn't match. Calculated: 44974ede47ec7db75852725c3c62b4a6d720025af39d0b426c7ca1d78698b7c8(sha256)  Expected: 824505ad6cb0fec0b5a38b488d3e331098fd4ffbd1d955629d1e827b8ed98c76(sha256) 
[MIRROR] slirp4netns-1.1.4-1.el8.x86_64.rpm: Downloading successful, but checksum doesn't match. Calculated: 44974ede47ec7db75852725c3c62b4a6d720025af39d0b426c7ca1d78698b7c8(sha256)  Expected: 824505ad6cb0fec0b5a38b488d3e331098fd4ffbd1d955629d1e827b8ed98c76(sha256) 
[FAILED] slirp4netns-1.1.4-1.el8.x86_64.rpm: No more mirrors to try - All mirrors were already tried without success  
```

**Describe the results you expected:**

I would have expeced the command `podman build -t centospodman .` to succeed.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
Version:      2.0.5
API Version:  1
Go Version:   go1.13.4
Built:        Thu Jan  1 01:00:00 1970
OS/Arch:      linux/amd64
```

**Output of `podman info --debug`:**

```
host:
  arch: amd64
  buildahVersion: 1.15.1
  cgroupVersion: v1
  conmon:
    package: conmon-2.0.20-1.el8.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.0.20, commit: 7df22a6bc0149d17c62f57027e0fe6d47d613f94'
  cpus: 16
  distribution:
    distribution: '"centos"'
    version: "8"
  eventLogger: file
  hostname: *** I removed this information /Erik ***
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1008
      size: 1
    - container_id: 1
      host_id: 427680
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1008
      size: 1
    - container_id: 1
      host_id: 427680
      size: 65536
  kernel: 4.18.0-193.14.2.el8_2.x86_64
  linkmode: dynamic
  memFree: 51691040768
  memTotal: 67206242304
  ociRuntime:
    name: runc
    package: runc-1.0.0-145.rc91.git24a3cf8.el8.x86_64
    path: /usr/bin/runc
    version: 'runc version spec: 1.0.2-dev'
  os: linux
  remoteSocket:
    path: /run/user/1008/podman/podman.sock
  rootless: true
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.1.4-1.el8.x86_64
    version: |-
      slirp4netns version 1.1.4
      commit: b66ffa8e262507e37fca689822d23430f3357fe8
      libslirp: 4.3.1
      SLIRP_CONFIG_VERSION_MAX: 2
  swapFree: 33810280448
  swapTotal: 33810280448
  uptime: 375h 36m 25.07s (Approximately 15.62 days)
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - registry.centos.org
  - docker.io
store:
  configFile: /home/test/.config/containers/storage.conf
  containerStore:
    number: 10
    paused: 0
    running: 1
    stopped: 9
  graphDriverName: overlay
  graphOptions:
    overlay.mount_program:
      Executable: /usr/bin/fuse-overlayfs
      Package: fuse-overlayfs-1.1.2-1.el8.x86_64
      Version: |-
        fusermount3 version: 3.2.1
        fuse-overlayfs: version 1.1.0
        FUSE library version 3.2.1
        using FUSE kernel interface version 7.26
  graphRoot: /home/test/.local/share/containers/storage
  graphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "false"
  imageStore:
    number: 56
  runRoot: /run/user/1008/containers
  volumePath: /home/test/.local/share/containers/storage/volumes
version:
  APIVersion: 1
  Built: 0
  BuiltTime: Thu Jan  1 01:00:00 1970
  GitCommit: ""
  GoVersion: go1.13.4
  OsArch: linux/amd64
  Version: 2.0.5
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
podman-2.0.5-1.el8.x86_64
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide?**

Yes

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by afbjorklund** [comment]
Looks like multiple builds of the same package, from different hosts (and with different checksums)

https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/CentOS_8/repodata/

```xml
<package type="rpm">
<name>slirp4netns</name>
<arch>x86_64</arch>
<version epoch="0" ver="1.1.4" rel="1.el8"/>
<checksum type="sha256" pkgid="YES">
824505ad6cb0fec0b5a38b488d3e331098fd4ffbd1d955629d1e827b8ed98c76
</checksum>
<summary>slirp for network namespaces</summary>
<description>
slirp for network namespaces, without copying buffers across the namespaces.
</description>
<packager/>
<url>https://github.com/rootless-containers/slirp4netns</url>
<time file="1598291245" build="1598291243"/>
<size package="52900" installed="91076" archive="92544"/>
<location href="x86_64/slirp4netns-1.1.4-1.el8.x86_64.rpm"/>
<format>
<rpm:license>GPLv2</rpm:license>
<rpm:vendor>obs://build.opensuse.org/devel:kubic</rpm:vendor>
<rpm:group>Unspecified</rpm:group>
<rpm:buildhost>cloud124</rpm:buildhost>
<rpm:sourcerpm>slirp4netns-1.1.4-1.el8.src.rpm</rpm:sourcerpm>
<rpm:header-range start="5096" end="14736"/>
<rpm:provides>
<rpm:entry name="slirp4netns" flags="EQ" epoch="0" ver="1.1.4" rel="1.el8"/>
<rpm:entry name="slirp4netns(x86-64)" flags="EQ" epoch="0" ver="1.1.4" rel="1.el8"/>
</rpm:provides>
<rpm:requires>
<rpm:entry name="libglib-2.0.so.0()(64bit)"/>
<rpm:entry name="libpthread.so.0()(64bit)"/>
<rpm:entry name="libpthread.so.0(GLIBC_2.2.5)(64bit)"/>
<rpm:entry name="libseccomp.so.2()(64bit)"/>
<rpm:entry name="libslirp.so.0()(64bit)"/>
<rpm:entry name="libslirp.so.0(SLIRP_4.0)(64bit)"/>
<rpm:entry name="libslirp.so.0(SLIRP_4.1)(64bit)"/>
<rpm:entry name="rtld(GNU_HASH)"/>
<rpm:entry name="libc.so.6(GLIBC_2.17)(64bit)"/>
</rpm:requires>
<file>/usr/bin/slirp4netns</file>
</format>
</package>
```

https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/CentOS_8/x86_64/

```
$ sha256sum slirp4netns-1.1.4-1.el8.x86_64.rpm 
44974ede47ec7db75852725c3c62b4a6d720025af39d0b426c7ca1d78698b7c8  slirp4netns-1.1.4-1.el8.x86_64.rpm
$ LANG=C rpm -qpi slirp4netns-1.1.4-1.el8.x86_64.rpm 
warning: slirp4netns-1.1.4-1.el8.x86_64.rpm: Header V3 RSA/SHA256 Signature, key ID 75060aa4: NOKEY
Name        : slirp4netns
Version     : 1.1.4
Release     : 1.el8
Architecture: x86_64
Install Date: (not installed)
Group       : Unspecified
Size        : 91076
License     : GPLv2
Signature   : RSA/SHA256, Thu Aug 20 15:53:16 2020, Key ID 4d64390375060aa4
Source RPM  : slirp4netns-1.1.4-1.el8.src.rpm
Build Date  : Thu Aug 20 15:53:14 2020
Build Host  : lamb60
Relocations : (not relocatable)
Vendor      : obs://build.opensuse.org/devel:kubic
URL         : https://github.com/rootless-containers/slirp4netns
Summary     : slirp for network namespaces
Description :
slirp for network namespaces, without copying buffers across the namespaces.

```

**Comment by lsm5** [comment]
OBS is stuck atm, but i'll rebuild a new one asap.

**Comment by lsm5** [comment]
should be fixed now. Closing this, please reopen if unfixed on your side.

**Comment by lsm5** [comment]
slirp4netns-1.1.4-2 and libslirp-4.3.1-2 are the builds with the fix

**Comment by lsm5** [comment]
fixed with `conmon-2.0.20-3`. Please reopen if you still notice this.

**Comment by lz006** [comment]
thanks, but now unfortunately it's slirp4netns again
```
[SKIPPED] container-selinux-2.144.0-3.el8.noarch.rpm: Already downloaded                                                                                                                                                                                                                                                     
[SKIPPED] catatonit-0.1.5-1.el8.x86_64.rpm: Already downloaded                                                                                                                                                                                                                                                               
[SKIPPED] containers-common-1.1.1-3.el8.x86_64.rpm: Already downloaded                                                                                                                                                                                                                                                       
(4/11): conmon-2.0.20-3.el8.x86_64.rpm                                                                                                                                                                                                                                                        121 kB/s |  42 kB     00:00    
(5/11): libslirp-4.3.1-1.el8.x86_64.rpm                                                                                                                                                                                                                                                       139 kB/s |  68 kB     00:00    
(6/11): libvarlink-util-19-3.el8.x86_64.rpm                                                                                                                                                                                                                                                   189 kB/s |  51 kB     00:00    
(7/11): podman-plugins-2.0.6-1.el8.x86_64.rpm                                                                                                                                                                                                                                                 3.7 MB/s | 2.3 MB     00:00    
(8/11): runc-1.0.0-145.rc91.git24a3cf8.el8.x86_64.rpm                                                                                                                                                                                                                                          11 MB/s | 5.8 MB     00:00    
[MIRROR] slirp4netns-1.1.4-1.el8.x86_64.rpm: Interrupted by header callback: Server reports Content-Length: 52868 but expected size is: 53056                                                                                                                                                                                
[MIRROR] slirp4netns-1.1.4-1.el8.x86_64.rpm: Interrupted by header callback: Server reports Content-Length: 52868 but expected size is: 53056                                                                                                                                                                                
[MIRROR] slirp4netns-1.1.4-1.el8.x86_64.rpm: Interrupted by header callback: Server reports Content-Length: 52868 but expected size is: 53056                                                                                                                                                                                
[MIRROR] slirp4netns-1.1.4-1.el8.x86_64.rpm: Interrupted by header callback: Server reports Content-Length: 52868 but expected size is: 53056                                                                                                                                                                                
[FAILED] slirp4netns-1.1.4-1.el8.x86_64.rpm: No more mirrors to try - All mirrors were already tried without success                                                                                                                                                                                                         
(10-11/11): containernetworking-plugins-0.8.7-1.el8.x86_64.rpm                                                                              60% [===================================================================================-                                                       ] 8.0 MB/s |  41 MB     00:03 ETA
The downloaded packages were saved in cache until the next successful transaction.
You can remove cached packages by executing 'dnf clean packages'.
Error: Error downloading packages:
  Cannot download x86_64/slirp4netns-1.1.4-1.el8.x86_64.rpm: All mirrors were tried

```

**Comment by lsm5** [comment]
@lz006 could you try a `dnf clean all` and then retry installing podman. I can install slirp4netns correctly on a clean centos8 env.

**Comment by lz006** [comment]
hi, you were right. It must be caused by some kind of tmp files. It is working. Thanks for quick response.

**Comment by mirandadam** [comment]
Hello, having a similar problem here on CentOS 8. Should I open a new issue? It seems there is some still undiscovered underlying motive for those breakages, since they keep coming back.

Excerpt of error message follows. Full output can be provided if requested.

```
# dnf -y module disable container-tools
...
# dnf -y install 'dnf-command(copr)'
...
# dnf -y copr enable rhcontainerbot/container-selinux
...
# curl -L -o /etc/yum.repos.d/devel:kubic:libcontainers:stable.repo https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/CentOS_8/devel:kubic:libcontainers:stable.repo
# dnf clean all
31 files removed
# dnf install -y podman
...
Total download size: 69 M
Installed size: 190 M
Downloading Packages:
(1/11): container-selinux-2.145.0-1.el8.noarch.rpm                                                       351 kB/s |  59 kB     00:00    
[MIRROR] catatonit-0.1.5-1.el8.x86_64.rpm: Interrupted by header callback: Server reports Content-Length: 295168 but expected size is: 295172
[MIRROR] containernetworking-plugins-0.8.7-1.el8.x86_64.rpm: Interrupted by header callback: Server reports Content-Length: 39442108 but expected size is: 39450796
[MIRROR] conmon-2.0.21-1.el8.x86_64.rpm: Downloading successful, but checksum doesn't match. Calculated: bee4679fab08b7fc4916957330650d687136bfdffbc4e8fab743510408906f63(sha256)  Expected: b707379ef905a11e8f017ce15ddb613f632d1c11f1594dbe29735292e17e6ad7(sha256) 
[MIRROR] catatonit-0.1.5-1.el8.x86_64.rpm: Interrupted by header callback: Server reports Content-Length: 295168 but expected size is: 295172
[FAILED] catatonit-0.1.5-1.el8.x86_64.rpm: No more mirrors to try - All mirrors were already tried without success                      
(3-4/11): conmon-2.0.21-1.el8.x86_64.rpm           0% [                                                ] 398 kB/s | 347 kB     02:56 ETA
The downloaded packages were saved in cache until the next successful transaction.
You can remove cached packages by executing 'dnf clean packages'.
Error: Error downloading packages:
  Cannot download x86_64/catatonit-0.1.5-1.el8.x86_64.rpm: All mirrors were tried
```

**Comment by mirandadam** [comment]
Update: The Centos_8_stream repo works fine. Centos_8 (not stream) gives above error.

**Comment by lsm5** [comment]
@mirandadam I'm able to install podman on a fresh centos_8 (Not stream) just fine using the install steps. 

If a `dnf clean all` alone doesn't help, I'd suggest this:
```
# dnf clean all
# rm -rf /var/cache/dnf
# rpmdb --rebuilddb
# dnf -y install podman
```
 
If that doesn't help either, I guess this is probably an OBS issue and would need to be reported there.

**Comment by lsm5** [comment]
Closing..

**Comment by mirandadam** [comment]
Indeed, I can no longer reproduce the problem today. I did try "dnf clean all" yesterday before my first post, and I also tried with a fresh vm, I should have mentioned that. Also CentOS_8_stream doesn't work today anymore and gives "Error: Unable to find a match: podman".

Thanks for the super fast reply and the troubleshooting tips, @lsm5 ! If it happens again I will open an issue as you suggested. Just for clarification, what is OBS and where is it's repository so I could report issues?
